src/net_processing.cpp

```diff
@@ -855,8 +852,9 @@ void PeerManager::FinalizeNode(const CNode& node, bool& fUpdateConnectionTime) {
     if (state->fSyncStarted)
         nSyncStarted--;
 
-    if (misbehavior == 0 && state->fCurrentlyConnected && !node.IsBlockOnlyConn()) {
-        // Note: we avoid changing visible addrman state for block-relay-only peers
+    if (node.fSuccessfullyConnected && misbehavior == 0 &&
+        !node.IsBlockOnlyConn() && !node.IsInboundConn()) {
+        // Only change visible addrman state for outbound, full-relay peers
```

Comment:
Perhaps it's worth explicitly documenting that we do not call Connected() for FEELER connections?  (This behavior emerges from feelers not having the `fSuccessfullyConnected` bool set, and I think it's desirable to maintain this behavior because otherwise long-lived block-relay-only connections might stick out as the ones which haven't been updated in a while.)

Reply:
Good suggestion. I'm touching this line again in https://github.com/bitcoin/bitcoin/pull/20228, which will need to be rebased once this is merged. Rather than invalidate the ACKs here for a comment update, I'll make the change there.

Reply:
Done in #20228.